sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_arg_properties.asciidoc

```diff
@@ -0,0 +1,547 @@
+struct conduit_key {
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    conduit_key>;
+};
+
+struct register_map_key {
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    register_map_key>;
+};
+
+struct stable_key {
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    stable_key>;
+};
+
+struct buffer_location_key {
+  template<int K>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    buffer_location_key, std::integral_constant<int, K>>;
+};
+
+struct awidth_key {
+  template<int K>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    awidth_key, std::integral_constant<int, K>>;
+};
+
+struct dwidth_key {
+  template<int K>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    dwidth_key, std::integral_constant<int, K>>;
+};
+
+enum class read_write_mode_enum {
+  read_write,
+  read,
+  write
+};
+
+struct read_write_mode_key {
+  template<read_write_mode_enum mode>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    read_write_mode_key, std::integral_constant<read_write_mode_enum, mode>>;
+};
+
+struct latency_key {
+  template<int K>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    latency_key, std::integral_constant<int, K>>;
+};
+
+struct maxburst_key {
+  template<int K>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    maxburst_key, std::integral_constant<int, K>>;
+};
+
+struct wait_request_key {
+  template<bool K>
+  using value_t = sycl::ext::oneapi::experimental::properties::property_value<
+    wait_request_key, std::integral_constant<bool, K>>;
+};
+
+inline constexpr conduit_key::value_t
+ conduit;
+inline constexpr register_map_key::value_t
+ register_map;
+inline constexpr stable_key::value_t stable;
+template<int K> inline constexpr buffer_location_key::value_t<K>
+ buffer_location;
+template<int K> inline constexpr awidth_key::value_t<K>
+ awidth;
+template<int K> inline constexpr dwidth_key::value_t<K>
+ dwidth;
+template<read_write_mode_enum mode>
+inline constexpr read_write_mode_key::value_t<mode>
+ read_write_mode;
+inline constexpr read_write_mode_key::value_t<
+  read_write_mode_enum::read>  read_write_mode_read;
+inline constexpr read_write_mode_key::value_t<
+  read_write_mode_enum::write>  read_write_mode_write;
+inline constexpr read_write_mode_key::value_t<
+  read_write_mode_enum::read_write>
+    read_write_mode_readwrite;
+template<int K> inline constexpr latency_key::value_t<K>
+  latency;
+template<int K> inline constexpr maxburst_key::value_t<K>
+ maxburst;
+template<int K> inline constexpr wait_request_key::value_t<K>
+ wait_request;
+inline constexpr wait_request_key::value_t<true>
+ wait_request_requested;
+inline constexpr wait_request_key::value_t<false>
+ wait_request_not_requested;
```

Comment:
I find the name `wait_request_not_requested` really confusing, and I don't think this convenience type helps.

The second use of "requested" doesn't seem to be tied to the first use of "request", and the description of `wait_request` later in the document talks about whether the wait request signal is generated rather than requested. `wait_request<true>` is also shorter.